src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBarracks.java

```diff
@@ -0,0 +1,198 @@
+                tempLevel += 1;
+                break;
+            case 3:
+                tempLevel += 2;
+                break;
+            default:
+                //do nothing
+        }
+
+        if (tempLevel > 4)
+        {
+            tempLevel -= 4;
+        }
+
+        EnumFacing facing = EnumFacing.NORTH;
+        switch (tempLevel)
+        {
+            case 1:
+                position = position.offset(EnumFacing.SOUTH, TOWER_OFFSET).offset(EnumFacing.WEST, TOWER_OFFSET);
+                break;
+            case 2:
+                position = position.offset(EnumFacing.NORTH, TOWER_OFFSET).offset(EnumFacing.EAST, TOWER_OFFSET);
+                facing = EnumFacing.SOUTH;
+                break;
+            case 3:
+                position = position.offset(EnumFacing.SOUTH, TOWER_OFFSET).offset(EnumFacing.EAST, TOWER_OFFSET);
+                facing = EnumFacing.WEST;
+                break;
+            case 4:
+                position = position.offset(EnumFacing.NORTH, TOWER_OFFSET).offset(EnumFacing.WEST, TOWER_OFFSET);
+                facing = EnumFacing.EAST;
+                break;
+            default:
+                //do nothing
+        }
+
+        return new Tuple<>(position, facing);
+    }
+
+    @Override
+    public void onDestroyed()
+    {
+        final World world = getColony().getWorld();
+        if (world != null)
+        {
+            for (int i = 1; i <= this.getBuildingLevel(); i++)
+            {
+                final Tuple<BlockPos, EnumFacing> tuple = getPositionAndFacingForLevel(i);
+                world.setBlockState(tuple.getFirst(), ModBlocks.blockHutBarracksTower.getDefaultState().withProperty(BlockHutBarracksTower.FACING, tuple.getSecond()));
+                getColony().addNewBuilding((TileEntityColonyBuilding) world.getTileEntity(tuple.getFirst()));
+            }
+        }
+        super.onDestroyed();
+    }
+
+    /**
+     * Gets the name of the schematic.
+     *
+     * @return Barracks schematic name.
+     */
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return BARRACKS;
+    }
+
+    /**
+     * Gets the max level of the Barracks's hut.
+     *
+     * @return The max level of the Barracks's hut.
+     */
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return BARRACKS_HUT_MAX_LEVEL;
+    }
+
+    /**
+     * BuildingDeliveryman View.
+     */
+    public static class View extends AbstractBuildingHut.View
+    {
+        /**
+         * Instantiate the deliveryman view.
+         *
+         * @param c the colonyview to put it in
+         * @param l the positon
+         */
+        public View(final ColonyView c, final BlockPos l)
+        {
+            super(c, l);
+        }
+
+        @NotNull
+        @Override
+        public Window getWindow()
+        {
+            return new WindowBarracksBuilding(this);
+        }
+
+        @Override
+        public void deserialize(@NotNull final ByteBuf buf)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method to simply inherit it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1185)